vaadin-icons-flow-parent/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/VaadinIcon.java

```diff
@@ -19,22 +19,21 @@
UTION, INVOICE, ITALIC, KEY_O, KEY, KEYBOARD_O, KEYBOARD, LAPTOP, LAYOUT, LEVEL_DOWN_BOLD, LEVEL_DOWN, LEVEL_LEFT_BOLD, LEVEL_LEFT, LEVEL_RIGHT_BOLD, LEVEL_RIGHT, LEVEL_UP_BOLD, LEVEL_UP, LIFEBUOY, LIGHTBULB, LINE_BAR_CHART, LINE_CHART, LINE_H, LINE_V, LINES_LIST, LINES, LINK, LIST_OL, LIST_SELECT, LIST_UL, LIST, LOCATION_ARROW_CIRCLE_O, LOCATION_ARROW_CIRCLE, LOCATION_ARROW, LOCK, MAGIC, MAGNET, MAILBOX, MALE, MAP_MARKER, MARGIN_BOTTOM, MARGIN_LEFT, MARGIN_RIGHT, MARGIN_TOP, MARGIN, MEDAL, MEGAPHONE, MEH_O, MENU, MICROPHONE, MINUS_CIRCLE_O, MINUS_CIRCLE, MINUS_SQUARE_O, MINUS, MOBILE_BROWSER, MOBILE_RETRO, MOBILE, MODAL_LIST, MODAL, MONEY_DEPOSIT, MONEY_EXCHANGE, MONEY_WITHDRAW, MONEY, MOON_O, MOON, MORNING, MOVIE, MUSIC, MUTE, NATIVE_BUTTON, NEWSPAPER, NOTEBOOK, NURSE, OFFICE, OPEN_BOOK, OPTION_A, OPTION, OPTIONS, ORIENTATION, OUT, OUTBOX, PACKAGE, PADDING_BOTTOM, PADDING_LEFT, PADDING_RIGHT, PADDING_TOP, PADDING, PAINT_ROLL, PAINTBRUSH, PALETTE, PANEL, PAPERCLIP, PAPERPLANE_O, PAPERPLANE, PARAGRAPH, PASSWORD, PASTE, PAUSE, PENCIL, PHONE_LANDLINE, PHONE, PICTURE, PIE_BAR_CHART, PIE_CHART, PIGGY_BANK_COIN, PIGGY_BANK, PILL, PILLS, PIN_POST, PIN, PLAY_CIRCLE_O, PLAY_CIRCLE, PLAY, PLUG, PLUS_CIRCLE_O, PLUS_CIRCLE, PLUS_MINUS, PLUS_SQUARE_O, PLUS, POINTER, POWER_OFF, PRESENTATION, PRINT, PROGRESSBAR, PUZZLE_PIECE, PYRAMID_CHART, QRCODE, QUESTION_CIRCLE_O, QUESTION_CIRCLE, QUESTION, QUOTE_LEFT, QUOTE_RIGHT, RANDOM, RASTER_LOWER_LEFT, RASTER, RECORDS, RECYCLE, REFRESH, REPLY_ALL, REPLY, RESIZE_H, RESIZE_V, RETWEET, RHOMBUS, ROAD_BRANCH, ROAD_BRANCHES, ROAD_SPLIT, ROAD, ROCKET, ROTATE_LEFT, ROTATE_RIGHT, RSS_SQUARE, RSS, SAFE_LOCK, SAFE, SCALE_UNBALANCE, SCALE, SCATTER_CHART, SCISSORS, SCREWDRIVER, SEARCH_MINUS, SEARCH_PLUS, SEARCH, SELECT, SERVER, SHARE_SQUARE, SHARE, SHIELD, SHIFT_ARROW, SHIFT, SHOP, SIGN_IN_ALT, SIGN_IN, SIGN_OUT_ALT, SIGN_OUT, SIGNAL, SITEMAP, SLIDER, SLIDERS, SMILEY_O, SORT, SOUND_DISABLE, SPARK_LINE, SPECIALIST, SPINNER_ARC, SPINNER_THIRD, SPINNER, SPLINE_AREA_CHART, SPLINE_CHART, SPLIT_H, SPLIT_V, SPLIT, SPOON, SQUARE_SHADOW, STAR_HALF_LEFT_O, STAR_HALF_LEFT, STAR_HALF_RIGHT_O, STAR_HALF_RIGHT, STAR_O, STAR, START_COG, STEP_BACKWARD, STEP_FORWARD, STETHOSCOPE, STOCK, STOP_COG, STOP, STOPWATCH, STORAGE, STRIKETHROUGH, SUBSCRIPT, SUITCASE, SUN_DOWN, SUN_O, SUN_RISE, SUPERSCRIPT, SWORD, TAB_A, TAB, TABLE, TABLET, TABS, TAG, TAGS, TASKS, TAXI, TEETH, TERMINAL, TEXT_HEIGHT, TEXT_INPUT, TEXT_LABEL, TEXT_WIDTH, THIN_SQUARE, THUMBS_DOWN_O, THUMBS_DOWN, THUMBS_UP_O, THUMBS_UP, TICKET, TIME_BACKWARD, TIME_FORWARD, TIMER, TOOLBOX, TOOLS, TOOTH, TOUCH, TRAIN, TRASH, TREE_TABLE, TRENDING_DOWN, TRENDING_UP, TROPHY, TRUCK, TWIN_COL_SELECT, TWITTER_SQUARE, TWITTER, UMBRELLA, UNDERLINE, UNLINK, UNLOCK, UPLOAD_ALT, UPLOAD, USER_CARD, USER_CHECK, USER_CLOCK, USER_HEART, USER_STAR, USER, USERS, VAADIN_H, VAADIN_V, VIEWPORT, VIMEO_SQUARE, VIMEO, VOLUME_DOWN, VOLUME_OFF, VOLUME_UP, VOLUME, WALLET, WARNING, WORKPLACE, WRENCH, YOUTUBE_SQUARE, YOUTUBE;
```

Comment:
Done in https://github.com/vaadin/flow-components/pull/1785/commits/4ea25e4efd538300c4137baf3c81f05b73d979f1

Reply:
If I read the commit right the misspelled icon names now remain but they won't work, will they? Assuming a v14 app uses `VaadinIcons.BUSS`, in v21+ it'll keep compiling (with a deprecation warning) but the icon would disappear. That's still a breaking change, is not it?

What I'm looking for here is that there will be a deprecation warning during the build (and in the IDEs), but the app would still actually work as before.

My guess is that it would require adding some new logic either to the `VaadinIcon` enum or to the `Icon` class to apply a custom name mapping for a few selected icons (`BUSS` -> `vaadin:bus` instead of `BUSS` -> `vaadin:buss`)

Reply:
Overriding enum.name() doesn't seem to be a feasible option so I ended up with normalizing the given VaadinIcon https://github.com/vaadin/flow-components/pull/1785/commits/8f9d7cd5c87ea3a6de94f77b6883fed2b9ea9cfd

Note that this will work with `new Icon(VaadinIcon.BUSS);` but not with `new Icon("buss");`

Reply:
I sugget to bring these misspelled names in the web component as well then. But that would be a separate task.

Reply:
https://github.com/vaadin/web-components/issues/2108